File: packages/health_connector_hk_ios/ios/health_connector_hk_ios/Sources/health_connector_hk_ios/mappers/measurement_unit_mappers/EnergyMapper.swift
import Foundation
import HealthKit

extension EnergyDto {
    /// Converts this DTO to a HealthKit `HKQuantity`.
    func toHealthKit() -> HKQuantity {
        HKQuantity(unit: .kilocalorie(), doubleValue: kilocalories)
    }
}

extension HKQuantity {
    /// Converts this HealthKit quantity to an `EnergyDto`.
    ///
    /// Uses kilocalories as the transfer unit for consistency.
    func toEnergyDto() -> EnergyDto {
        let kilocalories = doubleValue(for: .kilocalorie())
        return EnergyDto(kilocalories: kilocalories)
    }
}

File: packages/health_connector_hk_ios/lib/src/mappers/measurement_unit_mappers/energy_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Energy, sinceV1_0_0;
import 'package:health_connector_hk_ios/src/pigeon/health_connector_hk_ios_api.g.dart'
    show EnergyDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Energy] to [EnergyDto].
@sinceV1_0_0
@internal
/// Converts [Energy] to [EnergyDto].
@sinceV1_0_0
@internal
extension EnergyToDto on Energy {
  EnergyDto toDto() {
    // Uses kilocalories as the transfer unit for consistency.
    return EnergyDto(kilocalories: inKilocalories);
  }
}

/// Converts [EnergyDto] to [Energy].
@sinceV1_0_0
@internal
extension EnergyDtoToDomain on EnergyDto {
  Energy toDomain() {
    return Energy.kilocalories(kilocalories);
  }
}
